Constrained_triangulation_3/doc/Constrained_triangulation_3/Constrained_triangulation_3.txt

```diff
@@ -0,0 +1,224 @@
+This example demonstrates how to build such a triangulation.
+
+\cgalExample{Constrained_triangulation_3/conforming_constrained_Delaunay_triangulation_3_from_soup.cpp }
+
+
+\subsection CT_3_example_ccdt_fpmap Build a Conforming Constrained Delaunay Triangulation with Known Face Patches
+
+When the user knows a priori the set of patch ids that he wants to attach to faces, this information
+can be used and kept valid through the construction of the conforming constrained
+Delaunay triangulation.
+
+The following example demonstrates how to detect surface patches separated by sharp edges
+and use this surface segmentation in the tetrahedrization process.
+
+When this parameter is used, the 3D triangulation constrained faces indices (available with
+`CGAL::Conforming_constrained_Delaunay_triangulation_cell_data_3::face_constraint_index() `
+)
+are set to the corresponding patch ids from this map.
+When it is not used, the face indices are set to different indices for each input polygon.
+
+\cgalExample{Constrained_triangulation_3/conforming_constrained_Delaunay_triangulation_3_fpmap.cpp}
+
+\cgalFigureRef{ccdt_fpmap} shows the input and output of this triangulation construction example.
+
+\cgalFigureBegin{ccdt_fpmap, ccdt_fpmap.png}
+Left: Input PLC;
+Middle-left: The corresponding conforming constrained Delaunay triangulation with one patch per input polygon;
+Middle-Right: Input PLC segmented according to sharp edges;
+Right: The corresponding conforming constrained Delaunay triangulation using the segmentation.
+\cgalFigureEnd
+<!-- #camera 5.67784 5.7193 -5.46546 0.830007 -0.459002 0.09482 -0.302349 -->
+
+
+\subsection CT_3_examples_preprocessing Preprocessing the Input Mesh for Conforming Constrained Delaunay Triangulations
+
+The input mesh must be a valid triangulated surface mesh, i.e., it must not have self-intersections.
+Numerous preprocessing functions are available in the \ref PkgPolygonMeshProcessing package _to enforce these preconditions_.
+
+\subsubsection CT_3_example_ccdt_autorefinement Autorefinement of the Input Mesh
+
+The following examples shows how autorefinement can be used to preprocess the input mesh,
+when it is self-intersecting.
+
+\cgalExample{Constrained_triangulation_3/ccdt_3_after_autorefinement.cpp }
+
+
+\subsubsection CT_3_example_self_intersecting_non_triangles Non-Triangulated Self-Intersecting Input Mesh
+
+The following example shows how to build a conforming constrained Delaunay triangulation from
+an input mesh that is not triangulated, and (possibly) self-intersecting.
+
+The function `CGAL::Polygon_mesh_processing::does_self_intersect()` is used to check for self-intersections,
+but this function can be used only on triangulated meshes. It is therefore necessary to triangulate
+the input mesh before checking for self-intersections, using the function
+`CGAL::Polygon_mesh_processing::triangulate_faces()`.
+
+The triangulated version of the mesh is then kept in case of self-intersections,
```

Comment:
It is no longer a mesh but a triangle soup because of non-manifold edges. The mesh with duplicated edges (to resolve non-manifoldness) is not a problem for CDT3 but only for self-intersection test functions. If we still want to allow that, we need to use another self-intersection test function (like mesh -> soup -> merge duplicated points -> does_soup_self_intersect)